cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -191,17 +191,15 @@ private void validateCustomMapping(NamespaceMeta metadata) throws Exception {
       // if hbase namespace is provided validate no other existing namespace is mapped to it
       if (!Strings.isNullOrEmpty(metadata.getConfig().getHbaseNamespace()) &&
         metadata.getConfig().getHbaseNamespace().equals(existingConfig.getHbaseNamespace())) {
-        throw new NamespaceAlreadyExistsException(existingNamespaceMeta.getNamespaceId().toId(),
-                                                  String.format("A namespace '%s' already exists with the given " +
+        throw new IllegalArgumentException(String.format("A namespace '%s' already exists with the given " +
```

Comment:
what status code does the REST API return due to this?
500? 400?
Can you try it out?